pkg/controller/service/servicecontroller.go

```diff
@@ -488,6 +491,9 @@ func (s *ServiceController) needsUpdate(oldService *api.Service, newService *api
 			return true
 		}
 	}
+	if !reflect.DeepEqual(oldService.Annotations, newService.Annotations) {
```

Comment:
it's a map that contains strings, one of which is a list. you could still have a user update the annotation to go from: src-range1, src-range2 to src-range2, src-range1 right? that will cause us to do what? this deepequal will fail, but will it get ignored higher up the stack? or is it ok to just re-create the rule since it's the exact same? or will there be some period inbetween when we have no rule?